modules/internal/StringCasts.chpl

```diff
@@ -56,6 +56,24 @@ module StringCasts {
     return false;
   }
 
+  // homogenous tuples of primitive type
+  operator :(x: ?k*?t, type s:string) where isPrimitiveType(t) && isHomogeneousTupleType(x.type) {
+    var ret = "(";
+    for param i in 0..#k {
+      if i != 0 then ret += ", ";
+      if (x[i].type == c_string) {
+        try! {
+          ret += createStringWithNewBuffer(x[i]);
+        }
+      }
+      else {
+        ret += x[i]:string;
+      }
+    }
+    ret += ")";
```

Comment:
I think this is missing the trailing comma for 1-tuples.

Reply:
Also, this presumably exists because we don't normally write c_strings in IO?

Reply:
I'm not 100% sure; I actually copied the body of this cast from here: https://github.com/chapel-lang/chapel/blob/0329b84ba996a290aac708f8619f6e61a9b7e2da/modules/internal/ChapelIOStringifyHelper.chpl#L52-L70

And figured the `createStringWithNewBuffer` route was more efficient for `c_string` than a cast.

Reply:
I can remove it if that branch doesn't seem helpful though.

Reply:
I think it's fine as-is. Normally when trying to print a ``c_string`` in conventional IO, you'd get a compiler error prompting you to cast to a string first. Since we're already casting the tuple to a string, I think it's sensible to do the same with c_string elements.

Reply:
Do you know if this logic/rationale is still relevant given the recent c_string discussions (and pending deprecation of c_string)? Would we want to consider something for the thing that replaces c_string (c_ptr(<something>))

Reply:
Good point, I think we can remove the `c_string` specific logic now that it will no longer be a user facing type.